cookbook/controller/error_pages.rst

```diff
@@ -5,172 +5,157 @@
+a special **exception page** with lots of debug information to help you quickly
+discover the root problem:
 
-Thus, error pages can be customized in different ways, depending on how
-much control you need:
+.. image:: /images/cookbook/controller/error_pages/exceptions-in-dev-environment.png
+   :alt: A typical exception page in the development environment
 
-#. :ref:`Use the default ExceptionController and create a few
-   templates that allow you to customize how your different error
-   pages look (easy); <use-default-exception-controller>`
+Since these pages contain a lot of sensitive internal information, Symfony won't
+display them in the production environment. Instead, it'll show a simple and
+generic **error page**:
 
-#. :ref:`Replace the default exception controller with your own
-   (intermediate). <custom-exception-controller>`
+.. image:: /images/cookbook/controller/error_pages/errors-in-prod-environment.png
+   :alt: A typical error page in the production environment
 
-#. :ref:`Use the kernel.exception event to come up with your own
-   handling (advanced). <use-kernel-exception-event>`
+Error pages for the production environment can be customized in different ways
+depending on your needs:
+
+#. If you just want to change the contents and styles of the error pages to match
+   the rest of your application, :ref:`override default error templates <use-default-exception-controller>`.
+
+#. If you also want to tweak the logic used by Symfony to generate error pages,
+   :ref:`override the default exception controller <custom-exception-controller>`.
+
+#. If you need total control of exception handling to execute your own logic
+   :ref:`use the kernel.exception event <use-kernel-exception-event>`.
 
 .. _use-default-exception-controller:
 
-Using the Default ExceptionController
--------------------------------------
+Overriding the Default Error Templates
+--------------------------------------
 
-By default, the ``showAction()`` method of the
-:class:`Symfony\\Bundle\\TwigBundle\\Controller\\ExceptionController`
-will be called when an exception occurs.
+By default, when an exception occurs, the ``showAction()`` method of the
+:class:`Symfony\\Bundle\\TwigBundle\\Controller\\ExceptionController` is called
+thanks to an event listener configured by the TwigBundle.
 
-This controller will either display an
-*exception* or *error* page, depending on the setting of the ``kernel.debug``
-flag. While *exception* pages give you a lot of helpful
-information during development, *error* pages are meant to be
-shown to the user in production.
+Then, the controller selects one of the templates defined in the
+``Resources/views/Exception`` directory of the TwigBundle to render the error
+page. If you browse that directory (usually located in
+``vendor/symfony/symfony/src/Symfony/Bundle/TwigBundle``) you'll find a lot of
+templates defined for different types of errors and content formats.
```

Comment:
maybe also mention that `error.*.twig` is used in prod and `exception.*.twig` in dev? (if I remember correctly)